crates/autopilot/src/infra/persistence/dto/order.rs

```diff
@@ -289,6 +295,42 @@ pub struct Quote {
     pub fee: U256,
 }
 
+#[serde_as]
+#[derive(Clone, Debug, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct OrderQuote {
+    #[serde_as(as = "HexOrDecimalU256")]
+    pub sell_amount: U256,
+    #[serde_as(as = "HexOrDecimalU256")]
+    pub buy_amount: U256,
+    #[serde_as(as = "HexOrDecimalU256")]
+    pub fee: U256,
+    pub solver: H160,
```

Comment:
Thoughts on reusing the existing Quote dto and simply adding solver there? Generally, I think it's ok to add fields to existing structs (unless you are parsing in strict mode it shouldn't cause any issues) and this wouldn't reduce our API docs quite significantly.

Reply:
The same question here. https://github.com/cowprotocol/services/pull/2855#discussion_r1707132132
Conversion from domain into persistence requires a solver field, which is stored in the optional order's quote.

Reply:
But you can add the solver field to the [existing quote struct](https://github.com/cowprotocol/services/blob/7278d8994f2d7abfc7b8133c020eba30a30e1a95/crates/autopilot/src/infra/persistence/dto/order.rs#L280-L290), can you not? Why do you need to introduce a new struct here? I don't see how the fact that the quote is optional on the order makes a difference here?

I'd like to avoid us having duplicate structs with very unclear semantic differences (what is logically the difference between a Quote and an OrderQuote 🤯)

Reply:
Removed all the OrderQuotes. I hope I didn't miss anything.